sdk/tools/cabman/dfp.cxx

```diff
@@ -564,7 +555,7 @@ bool CDFParser::SetDiskNumber(PDISK_NUMBER *List, ULONG Number, ULONG Value)
         DN = DN->Next;
     }
 
-    DN = (PDISK_NUMBER)malloc(sizeof(DISK_NUMBER));
+    DN = new DISK_NUMBER;
```

Comment:
You could get slightly fancier, and have a constructor with arguments that take the DiskNumber and the Number parameters, that are otherwise initialized just below.